department-of-veterans-affairs/va.gov-cms
# VAGOV-4311: Add custom menu perm module.

## Description
See #4311

## Testing done
Visual / behat / phpunit

## Screenshots
![image](https://user-images.githubusercontent.com/2404547/110260962-542f7280-7f7c-11eb-96be-d56825c4e1d7.png)
![image](https://user-images.githubusercontent.com/2404547/110260974-58f42680-7f7c-11eb-8947-690904c647da.png)
![image](https://user-images.githubusercontent.com/2404547/110260983-5db8da80-7f7c-11eb-8649-32d99c9b65b6.png)
![image](https://user-images.githubusercontent.com/2404547/110260989-627d8e80-7f7c-11eb-98a9-cfa74111cae7.png)
![image](https://user-images.githubusercontent.com/2404547/110260993-66a9ac00-7f7c-11eb-99ef-911fb4f465bd.png)
![image](https://user-images.githubusercontent.com/2404547/110261002-6c06f680-7f7c-11eb-82d2-44a457676a32.png)
![image](https://user-images.githubusercontent.com/2404547/110262154-c0ac7080-7f80-11eb-84f2-3cc9845f249c.png)
![image](https://user-images.githubusercontent.com/2404547/110262160-c4d88e00-7f80-11eb-9351-8d4ee688dad4.png)


## QA steps
- As an administrator, go to `admin/config/content/va_gov_menu_access`
- Check the following roles: `Content creator - VAMC` & `Content editor`
- Check the following menus: `VA Butler health care` & `VA Bay Pines health care`
- Enter the following paths (one per line): `/butler-health-care/pharmacy` & `*/work-with-us`
- Go to `/user/1730/roles` and check the following roles: `Content creator - VAMC` , `Content editor` , `Content reviewer` , `Content publisher`
- Go to `user/1730/workbench_access` and check `VA Altoona health care` & `VA Butler health care`
- Sign out, and login as vanessa.luxen@va.gov
- Go to `/node/2551/edit` click save, and visually verify that you cannot save, and see `The path alias you have created has already been reserved.` message
- Go to `/node/2560/edit` click save, and visually verify that you cannot save, and see `The path alias you have created has already been reserved.` message
- Go to `/node/1332/edit` and visually verify that you see an access denied message
- Go to `node/add/health_care_region_detail_page` open up the menu settings, and visually verify that only VA Butler vamc items are available, and all other vamc items are greyed out and unclickable
- Go to `/admin/structure/menu/` and visually verify access is denied.
- Go to `/admin/content?title=&type=All&moderation_state=All&owner=185` and visually verify you can only edit Detail pages.

### Definition of Done


### Select Team for PR review


### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
